### PR TITLE
Upload CSV to partner portal

### PR DESCRIPTION
**Pull request by LDannijs**
#### Summary

Add a script, like the device repository, to sync the contents of this repo to Airtable so it can be used with the Partner Portal.

#### Changes

- Add a script that converts all the templates into a csv
- Update workflow to automate syncing to airtable

#### Notes for Reviewers

@benolayinka let me know if this seems correct.

#### Checklist

- [ ] Scope: The referenced issue is addressed, there are no unrelated changes.
- [ ] Compatibility: The changes are backwards compatible, they don't break existing deployments.
- [x] Testing: The changes are tested.
- [ ] Documentation: Relevant documentation is added or updated.

### CODE REVIEW

File: .github/workflows/validate.yml [hunk 1]
```diff
@@ -24,3 +24,24 @@ jobs:
...
+      - name: Install Node.js dependencies
+        run: npm ci
+      - name: Create templates.csv
+        run: node bin/csv.js
+      - name: Upload to Airtable
+        run: |
+          curl -X POST ${{ secrets.AIRTABLE_WEBHOOKS_API_ENDPOINT }} \
+          -H "Authorization: Bearer ${{ secrets.AIRTABLE_WEBHOOKS_API_KEY }}" \
+          -H "Content-Type: text/csv" \
+          --data-binary "@bin/templates.csv"
```
Comment (KrishnaIyer) [inline .github/workflows/validate.yml:47]: Let's separate this out to a separate action. Ex: `upload-to-airtable.yml` and make that action run on a push to the `master` branch.

**Review by KrishnaIyer — CHANGES_REQUESTED** [review]
The `node_modules` folder is not supposed to be checked in to version control. Please add that to `.gitignore` and delete it here.

File: .github/workflows/upload-airtable.yml [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+        uses: actions/setup-node@v4
+        with:
+          node-version: '20'
+      - name: Install Node.js dependencies
+        run: npm ci
+      - name: Create templates.csv
+        run: node bin/csv.js
+      - name: Upload to Airtable
+        run: |
+          curl -X POST ${{ secrets.AIRTABLE_WEBHOOKS_API_ENDPOINT }} \
```
Comment (KrishnaIyer) [inline .github/workflows/upload-airtable.yml:25]: `-X` is unnecessary since `curl` implies it.

File: .github/workflows/validate.yml [hunk 3]
```diff
@@ -23,4 +23,4 @@ jobs:
           cd schema
           go build -o ../bin/validate ./cmd/validate.go
       - name: Validate webhook templates
-        run: ./bin/validate
+        run: ./bin/validate
```
Comment (KrishnaIyer) [inline .github/workflows/validate.yml:26]: New line at end of file. Please fix you editor.

File: .gitignore [hunk 4]
```diff
@@ -0,0 +1,2 @@
+bin/templates.csv
+bin/node_modules
```
Comment (KrishnaIyer) [inline .gitignore:2]: New line at end of file. Please fix you editor.

File: bin/csv.js [hunk 5]
```diff
@@ -0,0 +1,50 @@
+const fs = require('fs');
+const path = require('path');
+const yaml = require('yaml');
+const { createObjectCsvWriter } = require('csv-writer');
+
```
Comment (KrishnaIyer) [inline bin/csv.js:5]: Can you write some tests for this code?

Comment (LDannijs) [reply bin/csv.js:5]: What do you mean with tests? If the csv is correct?

Comment (KrishnaIyer) [reply bin/csv.js:5]: Yes

Comment (LDannijs) [reply bin/csv.js:5]: I have ran it through various CSV checkers and the CSV is valid

Comment (KrishnaIyer) [reply bin/csv.js:5]: Validity is meant for syntax. How do you ensure that your generated CSV is actually correct, that all the correct fields are added?

**Review by KrishnaIyer — APPROVED** [review]
I'm conditionally approving this out of necessity (as discussed on Slack). There are no tests to validate if `csv.js` file has all the correct fields. I trust that this is manually tested for now.

File: bin/csv.js (hunk 5 above)
Comment (LDannijs) [reply bin/csv.js:5]: I tested if the csv was correct with the same tools as when i did for the devices and it was correct, but i am noticing we dont need the logo so ill fix that.

**Comment by LDannijs** [comment]
I have manually tested it with the airtable Sync API, and all the fields look correct.

image

**Pull request merged**